src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -256,12 +255,11 @@ Namespace Microsoft.CodeAnalysis.Semantics
 
         Private Function CreateBoundAssignmentOperatorOperation(boundAssignmentOperator As BoundAssignmentOperator) As IOperation
             Dim kind = GetAssignmentKind(boundAssignmentOperator)
-            Dim isImplicit As Boolean = boundAssignmentOperator.WasCompilerGenerated
             If kind = OperationKind.CompoundAssignmentExpression Then
```

Comment:
>Dim isChecked As Boolean = False [](start = 16, length = 32)

This code for ```isChecked``` and ```isLifted``` doesn't look right. Shouldn't we pull this information from the binary operator? Please add tests that expose incorrect behavior.

Reply:
BoundAssignmentOperatior doesn't need to have the flag, that information is available on the binary operator associated with the compound assignment, see the code  

```Dim temporaryRight = DirectCast(Create(boundAssignmentOperator.Right), IBinaryOperatorExpression)```